scripts/quota_log.py

```diff
@@ -35,12 +47,19 @@ def disk_quota():
 
 def cpu_hours():
     current_time = datetime.datetime.now()
-
-    # script that runs on uppmax
-    uq = subprocess.Popen(["/sw/uppmax/bin/projinfo", '-q'], stdout=subprocess.PIPE)
+    try:
+        # script that runs on uppmax
+        uq = subprocess.Popen(["/sw/uppmax/bin/projinfo", '-q'], stdout=subprocess.PIPE)
+    except Exception, e:
+        logging.error(e.message)
+        exit()
```

Comment:
right. Just reraise the exception, and let python handle it.

On Mon, Sep 14, 2015 at 3:42 PM, Per Kraulis <notifications@github.com>
wrote:

> In scripts/quota_log.py
> <https://github.com/SciLifeLab/scilifelab/pull/482#discussion_r39394278>:
>
> > @@ -35,12 +47,19 @@ def disk_quota():
> >
> >  def cpu_hours():
> >      current_time = datetime.datetime.now()
> > -
> > -    # script that runs on uppmax
> > -    uq = subprocess.Popen(["/sw/uppmax/bin/projinfo", '-q'], stdout=subprocess.PIPE)
> > +    try:
> > +        # script that runs on uppmax
> > +        uq = subprocess.Popen(["/sw/uppmax/bin/projinfo", '-q'], stdout=subprocess.PIPE)
> > +    except Exception, e:
> > +        logging.error(e.message)
> > +        exit()
>
> Didn't actually know about Python exit()!
>
> The problem with it is that it is very imperialistic to do exit() like
> this. It really is dictatorship over anyone trying to use this code as a
> module. Much better to raise an exception for the user to deal with as
> he/she sees fit.
>
> os.exit(1) is slightly better since it sends an error state to the
> environment, as you say.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/SciLifeLab/scilifelab/pull/482/files#r39394278>.
>